### ISSUE TITLE
ERROR: INVALID

### ISSUE THREAD

**Issue by flarkin1993** [arjvik/RPN-Ti84#1]
I am having trouble installing the program onto my student's TI-84 PLUS CE with PYTHON. I don't run LINUX and tried to install a linux based command line like ubuntu, but all of the troubles I have dealt with are above my paygrade when it comes to downloading the right dependencies and such.

I then downloaded the program directly, RPN2ASM.8xp and installed through TI-CONNECT CE and I keep getting an ERROR: INVALID code. I am not sure what the problem is, but it is more than likely on my end due to my own ignorance on programming languages.

You are a lifesaver for making a program like this. The RPN calculator is unfortunately becoming harder and harder to find and my students are more comfortable with the TI-84. 

Any help would be greatly appreciated.

**Comment by arjvik** [comment]
Glad that somebody else found a use for my program!
nd happy to help 

I've never seen that issue b

**Comment by arjvik** [comment]
Glad that somebody else found a use for my program! I'm happy to help you install and run it.

I've never seen that issue before. Does it show up when you transfer the file via TI-Connect or when you attempt to run it?

What version of the Ti-84 OS are you using?

And lastly, the issue may be that you have the newer Ti-84+CE with Python. I believe the processor/OS on that device may be significantly different from the classic CE, which would make the compiled .8xp file incompatible. I'm sure there is a way to modify the compiler so it produces a valid executable for the CE with Python calculator.

Out of curiosity, what are you using RPN for in your school? Is it perhaps Texas UIL Calculator?

**Comment by arjvik** [comment]
Now that I think about it, the error message might also be due to a recent OS update banning Assembly programs on certain calculators, because of the alleged risk that they could be used to cheat?

If that's the case, you can use a "hack" like [arTIfiCE](https://yvantt.github.io/arTIfiCE/) (Disclaimer: I have never used it) to add back that functionality.

**Comment by flarkin1993** [comment]
Yes it is. I work at a smaller school district in East Texas. We have older
HP calculators that are from the 70s and are breaking constantly. We don't
really have the funds for newer rpn calculator just for uil calculator, so
I started looking for programs for the ti that replicate a rpn.

Students seem more comfortable with the TI interface anyways.

I am relatively new in coaching calculator. You have any tips?

Attached is the OS version and other info.

**Comment by flarkin1993** [comment]
Are jailbroken calculators allowed? I'll have to reread the rules again.

**Comment by flarkin1993** [comment]
Does your program have a square root ability?

**Comment by arjvik** [comment]
GitHub doesn't handle attachments in emails; try uploading them directly through the web interface.

**Comment by arjvik** [comment]
I wouldn't worry about the legality of using arTIfiCE in a competition; it's still an off-the-shelf calculator, just using a program that exploits an OS weakness. From the perspective of the rules, I would argue that it counts as any other program.

**Comment by arjvik** [comment]
And no, I believe square roots are one of the operations that I still need to add, along with scientific notation. You can still calculate them with `.5^` though.

This program was literally thrown together at the last minute last year... I was working on it the night before and the morning of TMSCA State. Hopefully I get the time to improve it this year and make it on par with the classic HPs that most competitors use.

**Comment by flarkin1993** [comment]
Of course, I understand. I'm just grateful that such a program exists.
Thank you for all your hard work.

**Comment by arjvik** [comment]
Any luck making arTIfiCE work?

**Comment by flarkin1993** [comment]
It did the trick. You are awesome, I appreciate all the help. I'm telling
my students NOT to update their calculators.

I'll have my students use it and try to find bugs for you, if you'd like.

Thanks again.

**Comment by arjvik** [comment]
Hi @flarkin1993,

I thought you might be interested to hear that I've released [a new version](https://github.com/arjvik/RPN-Ti84/releases/tag/2.1) of my RPN program!

I've added lots of previously missing functionality into it, most notably square roots and scientific notation entry. I've modified the keystrokes for a few operations to make them more in line with regular use of the Ti-84.

I've also created [a table of all the supported keys and what they do](https://github.com/arjvik/RPN-Ti84#keys).

Download [RPN2ASM.8xp](https://github.com/arjvik/RPN-Ti84/releases/download/2.1/RPN2ASM.8xp).

Let me know if everything works, and if you have any suggestions for features to add! Best of luck for the upcoming TMSCA and District competitions this year!
